week7/answers.sql

```diff
@@ -1,3 +1,306 @@
+    ON c.character_id = iv.character_id
+    INNER JOIN items i
+    ON iv.item_id = i.item_id
+    UNION 
+    SELECT c.character_id, c.name AS character_name, i.name AS item_name, i.armor , i.damage
+    FROM characters c
+    INNER JOIN equipped e
+    ON c.character_id = e.character_id
+    INNER JOIN items i
+    ON e.item_id = i.item_id 
+    ORDER BY character_name, item_name;
+    
+CREATE OR REPLACE VIEW team_items AS
+    SELECT t.team_id, t.name AS team_name, i.name AS item_name, i.armor , i.damage
+    FROM teams t
+    INNER JOIN team_members te
+    ON t.team_id = te.team_id
+    INNER JOIN characters c
+    ON c.character_id = te.character_id
+    INNER JOIN inventory iv
+    ON c.character_id = iv.character_id
+    INNER JOIN items i
+    ON iv.item_id = i.item_id
+    UNION 
+    SELECT t.team_id, t.name AS team_name, i.name AS item_name, i.armor , i.damage
+    FROM teams t
+    INNER JOIN team_members te
+    ON t.team_id = te.team_id
+    INNER JOIN characters c
+    ON c.character_id = te.character_id
+    INNER JOIN equipped e
+    ON c.character_id = e.character_id
+    INNER JOIN items i
+    ON e.item_id = i.item_id 
+    ORDER BY team_name, item_name;
+    
+DELIMITER ;;
+    CREATE FUNCTION armor_total(id INT UNSIGNED)
+    RETURNS INT UNSIGNED 
+    DETERMINISTIC 
+    BEGIN 
+    DEClARE character_armor INT UNSIGNED;
+    DEClARE equipped_armor INT UNSIGNED;
+    DECLARE total_armor INT UNSIGNED;
+
+    SELECT armor  INTO character_armor
+    FROM character_stats 
+    WHERE character_id = id;
+
+    SELECT SUM(i.armor) INTO equipped_armor 
+    FROM items i
+    LEFT OUTER JOIN equipped e
+    ON e.item_id = i.item_id
+    WHERE e.character_id = id;
+
+    SET total_armor = character_armor + equipped_armor;
+        RETURN total_armor;
+
+    END;; 
+     
+DELIMITER ;;
+
+CREATE PROCEDURE attack(attacked_character_id INT UNSIGNED, equipped_id INT UNSIGNED)
+
+BEGIN
+DECLARE armor INT UNSIGNED;
+DECLARE damage INT UNSIGNED;
+DECLARE total_damage INT;
+DECLARE new_health INT;
+DECLARE total_health INT;
+
+
+SELECT armor_total(attacked_character_id) INTO armor;
+
+SELECT i.damage INTO damage 
+    FROM items i
+    	INNER JOIN equipped e
+    	ON e.item_id = i.item_id
+    WHERE e.equipped_id = equipped_id;
+SET total_damage = damage - armor;
+
+SELECT health INTO new_health
+    FROM character_stats
+    WHERE character_id = attacked_character_id ;
+
+IF total_damage > 0 THEN 
+
+    SET total_health = new_health - total_damage;
+    UPDATE character_stats SET health = total_health WHERE character_id = attacked_character_id;
+
+    IF  total_damage >= new_health THEN
+    DELETE FROM characters WHERE character_id = attacked_character_id;
+    END IF;
+END IF;
+END;;
+DELIMITER ;
+
+DELIMITER ;;
+CREATE PROCEDURE equip (item_inventory_id INT UNSIGNED)
+BEGIN
+DECLARE item INT UNSIGNED;
+DECLARE id_character INT UNSIGNED;
+DECLARE new_item INT UNSIGNED;
+DECLARE new_character INT UNSIGNED;
+
+SELECT item_id INTO item
```

Comment:
These four SELECT statements can be merged into one. For example:

```sql
SELECT character_id, item_id INTO new_character, new_item
FROM inventory
WHERE inventory_id = item_inventory_id;
```

You'll find that these statements are true:
`item == new_item`
`id_character == new_character`

(Don't forget to remove unused variable declarations after simplifying.)